### PR TITLE
[#1998] `useBem` hook adaptation for camel-case modifiers

### PR DESCRIPTION
**Pull request by aluarius**
Closes: #1998 

## Description
Now we can pass camel-case modifiers to `useBem` hook:

```vue
const computedClass = useBem('va-switch', () => ({
...pick(props, ['readonly', 'disabled', 'leftLabel']),
}))
```

instead of:

```vue
const computedClass = useBem('va-switch', () => ({
...pick(props, ['readonly', 'disabled']),
'left-label': props.leftLabel,
}))
```

to gain `va-switch--left-label`

P.S. 2000 pr! :)

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: packages/ui/src/components/va-switch/VaSwitch.vue [hunk 1]
```diff
@@ -160,12 +160,11 @@ export default defineComponent({
...
     const computedClass = useBem('va-switch', () => ({
-      ...pick(props, ['readonly', 'disabled']),
+      ...pick(props, ['readonly', 'disabled', 'leftLabel']),
       checked: isChecked.value,
       indeterminate: isIndeterminate.value,
       small: props.size === 'small',
       large: props.size === 'large',
-      'left-label': props.leftLabel,
       error: computedError.value,
       'on-keyboard-focus': hasKeyboardFocus.value,
```
Comment (RVitaly1978) [inline packages/ui/src/components/va-switch/VaSwitch.vue:169]:
```suggestion
      onKeyboardFocus: hasKeyboardFocus.value,
```

File: packages/ui/src/composables/useBem.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { computed, Ref, unref, ComputedRef } from 'vue'
 import { __DEV__ } from '../utils/global-utils'
 import isFunction from 'lodash/isFunction.js'
 
-type Key<Prefix extends string, ModifierKey extends string> = `${Prefix}--${ModifierKey}`
+type Key<Prefix extends string, ModifierKey extends string> = `${Prefix}--${ModifierKey | string}`
```
Comment (RVitaly1978) [inline packages/ui/src/composables/useBem.ts:5]:
```suggestion
type Key<Prefix extends string, ModifierKey extends string> = `${Prefix}--${ModifierKey}`
```

File: packages/ui/src/composables/useBem.ts [hunk 3]
```diff
@@ -34,13 +34,23 @@ export const useBem = <ModifierKey extends string, Prefix extends string>(
...
 
+  const isUppercase = (str: string) => str !== str.toLowerCase()
+  const getModifierKey = (key: string) => {
+    if (!isUppercase(key) || isUppercase(key.charAt(0))) { return key.toLowerCase() }
+
+    return key.split('').reduce((acc, char) => {
+      isUppercase(char) ? acc += `-${char.toLowerCase()}` : acc += char
+      return acc
+    }, '')
+  }
```
Comment (RVitaly1978) [inline packages/ui/src/composables/useBem.ts:45]: lodash.kebabCase ?

**Review by RVitaly1978 — CHANGES_REQUESTED** [review]

File: packages/ui/src/composables/useBem.ts (hunk 2 above)
Comment (aluarius) [reply packages/ui/src/composables/useBem.ts:5]: In case `ModifierKey` is `propName` (`prop-name` after kebab-cased) this will be incorrect, doesn't it?

**Review by rustem-nasyrov — APPROVED** [review]

**Review by m0ksem — APPROVED** [review]
Document it with jsdocs

**Pull request merged**